File: app/src/main/java/com/fieldservice/app/ui/components/PriorityBadge.java
package com.fieldservice.app.ui.components;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.widget.TextView;

import androidx.core.content.ContextCompat;
import androidx.core.graphics.ColorUtils;

import com.fieldservice.app.R;
import com.fieldservice.app.domain.model.Priority;

/** Selo de prioridade do chamado. Não depende só da cor: o texto identifica o nível. */
public final class PriorityBadge {

    private static final int BACKGROUND_ALPHA = (int) (0.16f * 255);

    private PriorityBadge() {
    }

    public static String label(Priority priority) {
        switch (priority) {
            case LOW:
                return "Baixa";
            case MEDIUM:
                return "Média";
            case HIGH:
                return "Alta";
            case CRITICAL:
                return "Crítica";
            default:
                return "";
        }
    }

    private static int colorRes(Priority priority) {
        switch (priority) {
            case LOW:
                return R.color.priority_low;
            case MEDIUM:
                return R.color.priority_medium;
            case HIGH:
                return R.color.priority_high;
            case CRITICAL:
                return R.color.priority_critical;
            default:
                return R.color.color_on_surface;
        }
    }

    public static void bind(TextView view, Priority priority) {
        Context context = view.getContext();
        int color = ContextCompat.getColor(context, colorRes(priority));

        view.setText(label(priority));
        view.setTextColor(color);

        Drawable background = ContextCompat.getDrawable(context, R.drawable.bg_badge);
        if (background != null) {
            background = background.mutate();
            background.setTint(ColorUtils.setAlphaComponent(color, BACKGROUND_ALPHA));
            view.setBackground(background);
        }
    }
}

File: legacy-kotlin/app/src/main/java/com/fieldservice/app/ui/components/PriorityBadge.kt
package com.fieldservice.app.ui.components

import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.fieldservice.app.domain.model.Priority
import com.fieldservice.app.ui.theme.PriorityCritical
import com.fieldservice.app.ui.theme.PriorityHigh
import com.fieldservice.app.ui.theme.PriorityLow
import com.fieldservice.app.ui.theme.PriorityMedium

private fun Priority.label(): String = when (this) {
    Priority.LOW -> "Baixa"
    Priority.MEDIUM -> "Média"
    Priority.HIGH -> "Alta"
    Priority.CRITICAL -> "Crítica"
}

private fun Priority.color(): Color = when (this) {
    Priority.LOW -> PriorityLow
    Priority.MEDIUM -> PriorityMedium
    Priority.HIGH -> PriorityHigh
    Priority.CRITICAL -> PriorityCritical
}

/** Selo de prioridade do chamado. Não depende só da cor: o texto identifica o nível. */
@Composable
fun PriorityBadge(priority: Priority, modifier: Modifier = Modifier) {
    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(6.dp),
        color = priority.color().copy(alpha = 0.12f),
        contentColor = priority.color()
    ) {
        Text(
            text = priority.label(),
            style = MaterialTheme.typography.labelMedium,
            modifier = Modifier.padding(horizontal = 10.dp, vertical = 4.dp)
        )
    }
}
